lib/rex/post/meterpreter/packet.rb

```diff
@@ -693,17 +819,31 @@ def to_r
   # passing it on to the default functionality that can parse
   # the TLV values.
   #
-  def from_r(bytes)
-    xor_key = bytes[0,4].unpack('N')[0]
-    super(xor_bytes(xor_key, bytes[4, bytes.length]))
+  def from_r(aes_key=nil)
+    xor_key = self.raw[XOR_KEY_OFF, XOR_KEY_SIZE]
+    data = xor_key + xor_bytes(xor_key, self.raw[SESSION_GUID_OFF, self.raw.length - SESSION_GUID_OFF])
+    self.session_guid = data[SESSION_GUID_OFF, SESSION_GUID_SIZE]
+    encrypted_flag = data[ENCRYPTED_FLAG_OFF, ENCRYPTED_FLAG_SIZE] == "\x01"
```

Comment:
To be honest, no. At the moment it's just a single bit. But it'll actually become an identifier which represents which encryption scheme is in use (such as something cheaper to run on smaller processors). So it'll stay as is for now :) Then later we'll probs change it to something indicative of the encryption scheme. Such as `ENCRYPTION_SCHEME` (where `00` will be none, and `01` will be AES256.. etc. Thanks!